internal/testingx/tcpx.go

```diff
@@ -0,0 +1,44 @@
+package testingx
+
+import "net"
+
+// TCPListener creates TCP connections for HTTP, TLS, etc. This type should work both
+// with the standard library and with netem as its backend.
+type TCPListener interface {
+	ListenTCP(network string, addr *net.TCPAddr) (net.Listener, error)
+}
+
+// TCPListenerStdlib implements [HTTPListener] for the stdlib.
+type TCPListenerStdlib struct{}
+
+var _ TCPListener = &TCPListenerStdlib{}
+
+// ListenTCP implements HTTPListener.
+func (*TCPListenerStdlib) ListenTCP(network string, addr *net.TCPAddr) (net.Listener, error) {
+	return net.ListenTCP(network, addr)
+}
+
+// tcpMaybeResetNetConn is a portable mechanism to reset a net.Conn that takes into account
+// both TLS wrapping with any library and stdlib vs. netem concerns.
```

Comment:
```suggestion
// both TLS wrapping with any library and stdlib vs. netem concerns.
//
// Bug: netem is not WAI because there's no *gonet.TCPConn.SetLinger method.
```